### DISCUSSION TITLE
Dynamically render react-components streamed from an LLM

### DISCUSSION THREAD

**Discussion by rajeshdavidbabu** [vercel/next.js#59485] [App Router]
Hey Team,

  **What am I doing ?**

  I am currently building an internal tool for my company where I would dynamically render React components generated using Chatgpt using React Server Components. 

 **The example I am using** 
  
  I am using this official ai-sdk server component example, https://sdk.vercel.ai/docs/guides/frameworks/nextjs-app#in-server-components
  
  **What do I want ?**
  
   I want something similar, but instead of rendering the text, _**I need to parse and render React component form the stream data**_ and return them. 
   
    Kinda similar to V0.dev's approach when we run the initial render.

**Comment by Kaidanov** [comment]
did you fined an answer?

**Comment by rajeshdavidbabu** [comment]
Currently you can do this already with AI SDK. Checkout https://www.morphic.sh/ and its open-source code-base. https://www.youtube.com/watch?v=XdaiGK3ySj0&t

**Comment by dnascimento** [comment]
Looking for the same but with Vite
